src/tool_xattr.c

```diff
@@ -49,6 +49,40 @@ static const struct xattr_mapping {
   { NULL,                  CURLINFO_NONE } /* last element, abort loop here */
 };
 
+/* returns TRUE if a new URL is returned, that then needs to be freed */
+static bool stripcredentials(char **url)
```

Comment:
This seems like a prime candidate for a unittest to ensure we don't break it at some point?